ceremonies/meetup-validation/src/lib.rs

```diff
@@ -38,15 +41,83 @@ pub fn get_participant_judgements(
 		n_confirmed,
 	)?);
 
+	// this check has to be made before we exclude participants based on attestations
+	let early_rewards_possible = early_rewards_possible(
+		participant_judgements.legit.clone(),
+		participant_attestations.clone(),
+		participants.len(),
+		n_confirmed,
+		vote_is_unanimous,
+	);
+
 	participant_judgements.exclude_participants(get_excluded_participants_num_attestations(
 		&participant_judgements.legit,
 		participant_attestations.clone(),
 		attestation_threshold_fn,
 	)?);
 
+	participant_judgements.early_rewards_possible = early_rewards_possible;
 	Ok(participant_judgements)
 }
 
+fn vote_yields_majority(num_participants: usize, n_confirmed: u32) -> bool {
+	n_confirmed as f64 > (num_participants as f64) / 2.0
+}
+
+fn num_attestations_matches_vote(
+	legit_participants: &Participants,
+	participant_attestations: &Attestations,
+	n_confirmed: u32,
+) -> bool {
+	participant_attestations
+		.into_iter()
+		.enumerate()
+		.map(|(i, v)| !(legit_participants.contains(&i)) || v.len() == (n_confirmed - 1) as usize)
+		.all(|item| item)
+}
+
+fn attestation_graph_is_fully_connected(
+	legit_participants: Participants,
+	participant_attestations: Attestations,
+) -> bool {
+	for (i, mut attestations) in participant_attestations.into_iter().enumerate() {
```

Comment:
what happens, if `i` attested people that fell out of the legit participant set? I believe your logic with sorting and testing equality would not see this as fully connected. but it should

Anyway, sorting twice within a for loop doesn't strike me as efficient. Wouldn't a nested for loop through legit participants be more efficient?

Reply:
this test is made before any participants can fall out of the set, because the attestation graph needs to be fully connected for all participants in order for early payout to be possible, right?

about the efficiency. my implementation:
O(n) (loop over attestations) * O(n^2) (sorting, worstcase)

nested loops:
O(n) (loop over attestations) * O(n) (loop over each elem in attestations) * O(n) (check if legit_participant contains elem)

so in asymptotic complexity they are the same.
i agree that in the first implementation we have some overhead because it is 2*O(n^2), but i would argue that for those small input sizes it shouldnt really matter.

but you decide, should i change it to the nested loop?

Reply:
nah, its bounded, so I guess it's fine. Do we already have a worst case benchmark for this claim function? that will tell us the price